### PR TITLE
[IMP][15.0] website: make translation components extensible

### PR DESCRIPTION
**Pull request by royle-vietnam**
It is not common, but we sometimes need to modify translation components in place. The goal is to have a mechanism to change translation components and all future/present instances

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by royleviindoo** [comment]
@aab-odoo sorry for bothering you, can you help me review this pr, thank you

**Comment by tranngocson1996** [comment]
@bso-odoo Are you in charge of this module? We are in dire need of it. If it's not you, please ping me to the responsible person. Thank you so much 👍🏿

**Comment by bso-odoo** [comment]
> @bso-odoo Are you in charge of this module? We are in dire need of it. If it's not you, please ping me to the responsible person. Thank you so much 👍🏿

@rdeodoo Maybe this should even be in 14.0 ?

File: addons/website/static/src/js/menu/translate.js [hunk 1]
```diff
@@ -350,4 +350,11 @@ registry.category("website_navbar_widgets").add("TranslatePageMenu", {
     Widget: TranslatePageMenu,
     selector: '.o_menu_systray:has([data-action="translate"])',
```
Comment (bso-odoo) [inline addons/website/static/src/js/menu/translate.js:351]: In the meantime, try to adapt your commit message to match the [guideline](https://www.odoo.com/documentation/15.0/contributing/development/coding_guidelines.html#tag-and-module-name).

E.g. `[IMP] website: make translation components extensible` and some explanation after a blank line to describe why this is helpful.

Comment (royleviindoo) [reply addons/website/static/src/js/menu/translate.js:351]: @bso-odoo okey, done

**Review by rdeodoo — APPROVED** [review]
> Maybe this should even be in 14.0 ?

Generally:
- We won't do anything in stable to help with that kind of stuff, especially in python. That's really rare we change code to make it easily extensible in stable. Nothing comes risk free + it would be hard to draw a line. Generally either it is a bug or an imp, and rule of thumb is: bug -> Stable, imp -> master
- We did it recently for the /shop controllers and the autocomplete behavior, but that's really exceptions as those are really some key code.
- For the present case, it doesn't really hurt and is quite minimal diff to do it, plus already provided by the community, so it's fine.

Really a case by case and common sense I guess :)
But we clearly don't want the community to start opening that kind of PR all over the code in stable. We can accept a few where it makes sense / is easy / provided by the community.

Note that we usually keep those kind of feedback in mind when refactoring in master.

@robodoo delegate=bso-odoo

**Comment by bso-odoo** [comment]
@robodoo r+
Thank you for your contribution !

**Pull request closed without merging**